### DISCUSSION TITLE
THREEJS version discrepancy between modelviewer.dev/editor and  @google/model-viewer npm package 2.1.1

### DISCUSSION THREAD

**Discussion by furkandindar** [google/model-viewer#4042] [Q&A] [answered]
Hello,

I am using latest model-viewer (2.1.1) on my app which is using three.js r146.
`
{
  "name": "@google/model-viewer",
  "version": "2.1.1",
"dependencies": {
    "three": "^0.146.0",
  }
}`
<img width="539" alt="Screen Shot 2023-01-10 at 2 22 23 PM" src="https://user-images.githubusercontent.com/33446963/211538342-c3f2b471-7f9a-4fe6-b330-0b7642418606.png">

But when I check [modelviewer.dev/editor](https://modelviewer.dev/editor/), it's using three.js r148
<img width="540" alt="Screen Shot 2023-01-10 at 2 22 02 PM" src="https://user-images.githubusercontent.com/33446963/211538241-299079b6-d320-459c-99e2-b32f8b95a6d8.png">

and as i see from [three.js release notes r148](https://github.com/mrdoob/three.js/releases/tag/r148) they have added `TwoPassDoubleSide`.

Because of this, the same model looks different on my model-viewer and modelviewer.dev/editor, here is the comparison;
my model-viewer (2.1.1) latest with three.js r146;
<img width="293" alt="Screen Shot 2023-01-10 at 2 32 26 PM" src="https://user-images.githubusercontent.com/33446963/211541608-bc583772-22eb-455f-82f5-b570d6f6c362.png">

[modelviewer.dev/editor/](https://modelviewer.dev/editor/) with three.js r148;
<img width="372" alt="Screen Shot 2023-01-10 at 2 33 54 PM" src="https://user-images.githubusercontent.com/33446963/211541843-31d814da-48bd-4366-8de0-ccd3ac84f638.png">

there is a difference between 2 modelviewers on the nose pad part (while looking through the glass), and on the rayban logo. 

My question is; am I right on my inference on why this happens (three.js r148 `TwoPassDoubleSide`), and when will you release a new version supporting three.js r148?

**Comment by elalish** [comment, marked as answer]
Indeed you are correct, as the editor and docs use pre-release MV. However, `TwoPassDoubleSide` also changed the default behavior (which is being corrected), but has not yet been corrected here. The next release will end up looking the same as v2.1.1.

**Reply by elalish** [reply]
Three.js fixed this upstream and has been merged here, so please verify this is working as expected again.
